src/lib.rs

```diff
@@ -146,16 +146,71 @@ fn url_to_file_path_wasm(url: &Url) -> Result<PathBuf, ()> {
 #[inline]
 pub fn normalize_path(path: Cow<Path>) -> Cow<Path> {
   fn should_normalize(path: &Path) -> bool {
+    if path_has_trailing_separator(path) {
+      return true;
+    }
+
     for component in path.components() {
       match component {
-        Component::Prefix(..) | Component::CurDir | Component::ParentDir => {
-          return true
+        Component::CurDir | Component::ParentDir => {
+          return true;
         }
-        Component::RootDir | Component::Normal(_) => {
+        Component::Prefix(..) | Component::RootDir | Component::Normal(_) => {
           // ok
         }
       }
     }
+
+    path_has_cur_dir_separator(path)
```

Comment:
Why not checking for `/../` too?

Reply:
`Component::ParentDir` will be hit above thankfully.